File: AtCoder/AtCoder Beginner Contest/085/C/C/ABC_085_C_01.c
#include <stdio.h>

int main(void)
{
    int N, Y;
    scanf("%d %d", &N, &Y);
    int i, j, k;
    int flg = 0;
    int result[3] = {-1, -1, -1};

    for (i = 0; i <= N && flg == 0; i++) {
        for (j = 0; j <= N && flg == 0; j++) {
            if ((k = N - (i + j)) < 0) {
                continue;
            }
            int sum = 10000*i + 5000*j + 1000*k;
            if (sum == Y) {
                flg = 1;
                result[0] = i;
                result[1] = j;
                result[2] = k;
                break;
            }
        }
    }
    printf("%d %d %d\n", result[0], result[1], result[2]);

    return 0;
}

File: AtCoder/AtCoder Beginner Contest/085/C/Go/ABC_085_C_01.go
package main

import "fmt"

func main() {
	var N, Y int
	fmt.Scanf("%d %d", &N, &Y)
	flg := 0
	result := [3]int{-1, -1, -1}

	for i := 0; i <= N && flg == 0; i++ {
		for j := 0; j <= N && flg == 0; j++ {
			k := N - (i + j)
			if k < 0 {
				continue
			}
			sum := 10000*i + 5000*j + 1000*k
			if (sum == Y) {
				flg = 1
				result[0] = i;
				result[1] = j;
				result[2] = k;
			}
		}
	}

	fmt.Printf("%d %d %d\n", result[0], result[1], result[2]);
}